### PR TITLE
Add ability to generate express app

### PR DESCRIPTION
**Pull request by grawk**
Makes generator more useful by allowing a fully self contained express app and nemo suite in a single step.

### CODE REVIEW

File: app/templates/_app.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			}, appTimeout);
+		});
+		bbt[0].addEventListener('click', function () {
+			result[0].innerHTML = "Please wait";
+			setTimeout(function () {
+				bn = document.querySelectorAll('#ban')[0].value;
+				result[0].innerHTML = "Please wait";
+				setTimeout(function () {
+					result[0].innerHTML = (bn === "1001001") ? "<p class='result bad'>Unable to add your bank account</p>" : "<p class='result good'>Successfully added your bank account</p>";
+					//result[0].innerHTML = "<p class='result good'>Successfully added your bank account</p>";
```
Comment (gabrielcsapo) [inline app/templates/_app.js]: do we need this commented out string?

File: app/templates/_server.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+app.use('/static', express.static('public'));
+app.get('/', function (req, res) {
+	res.sendFile(path.resolve(__dirname, 'public/index.html'));
+});
+
+var server = app.listen(3000, function () {
+	var host = server.address().address;
+	var port = server.address().port;
+
+	console.log('Example app listening at http://%s:%s', host, port);
```
Comment (gabrielcsapo) [inline app/templates/_server.js]: should we change `Example app` to pickup the name from the package.json?

`require('package.json').name`

File: app/index.js [hunk 3]
```diff
@@ -50,10 +57,18 @@ var NemoGenerator = yeoman.generators.Base.extend({
...
+      var destRoot = (props.existingServer) ? path.resolve(process.cwd()) : path.resolve(process.cwd(), props.newAppDir);
+      this.destinationRoot(destRoot);
       this.baseDirOption = props.baseDirOption;
       this.browserOption = props.browserOption;
       this.testFramework = props.testFramework;
+      this.existingServer = props.existingServer;
+      this.newAppDir = props.newAppDir;
       this.deployedUrl = (!!props.deployedUrl) ? props.deployedUrl : undefined;
+      this.baseUrl = (props.existingServer) ? 'https://fast-castle-8102.herokuapp.com' : 'http://localhost:3000';
+      //console.log(this.user);
```
Comment (gabrielcsapo) [inline app/index.js]: can we remove this commented line?

File: app/templates/_app.js (hunk 1 above)
Comment (grawk) [inline app/templates/_app.js]: Oh. No. Never. I'll remove that.

File: app/templates/_server.js (hunk 2 above)
Comment (grawk) [inline app/templates/_server.js]: Sure, why not?

File: app/index.js (hunk 3 above)
Comment (grawk) [inline app/index.js]: yes. will do.

File: app/templates/_server.js (hunk 2 above)
Comment (gabrielcsapo) [inline app/templates/_server.js]: we could even overwrite the file during compilation?

**Comment by grawk** [comment]
In trying to add a unit test for this, I got out in the weeds upgrading the test environment. Still working on it.

File: app/templates/_server.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+app.use('/static', express.static('public'));
+app.get('/', function (req, res) {
+	res.sendFile(path.resolve(__dirname, 'public/index.html'));
+});
+
+var server = app.listen(3000, function () {
+	var host = server.address().address;
+	var port = server.address().port;
+
+	console.log('<%= newAppDir %> listening at http://%s:%s', host, port);
```
Comment (gabrielcsapo) [inline app/templates/_server.js]: nice! 👍 it is definitely faster then looking at the `package.json` and getting the name of the project.

**Comment by grawk** [comment]
@gabrielcsapo do you give your overall blessing to this? I'm planning to publish a major.

**Comment by gabrielcsapo** [comment]
It has been blessed 👏🏻 @grawk

**Pull request merged**